backend/src/util/hash.rs

```diff
@@ -0,0 +1,57 @@
+use std::fmt::{self, Debug};
+
+use serde::de::{self, Deserialize, Deserializer, Unexpected, Visitor};
+
+const HASH_BYTES: usize = 32;
+
+/// Newtype warpper for 32-byte hash values, implementing readable `Debug` and `serde::Deserialize`.
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub struct Hash([u8; HASH_BYTES]);
```

Comment:
Merits a comment to our future selves: "We could use `primitive_types::H256` here, but opted for a custom type to aboid more dependencies."